src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Windows/Input/KeyConverter.cs

```diff
@@ -4,246 +4,326 @@
-                    return String.Empty;
-                }
-
-                if (key >= Key.D0 && key <= Key.D9)
-                {
-                    return Char.ToString((char)(int)(key - Key.D0 + '0'));
-                }
-
-                if (key >= Key.A && key <= Key.Z)
-                {
-                    return Char.ToString((char)(int)(key - Key.A + 'A'));
-                }
-
-                String strKey = MatchKey(key, culture);
-                if (strKey != null)
-                {
-                    return strKey;
-                }
-            }
-            throw GetConvertToException(value, destinationType);
+                Key.None => string.Empty,
+                // This is a fast path for common keys before resort to Enum<Key>.ToString()
+                >= Key.D0 and <= Key.D9 => char.ToString((char)(key - Key.D0 + '0')),
+                >= Key.A and <= Key.Z => char.ToString((char)(key - Key.A + 'A')),
+                // We format some keys differently than defined in the enum
+                Key.Back => "Backspace",
+                Key.LineFeed => "Clear",
+                Key.Escape => "Esc",
+                // We will add some heavily used interned strings (F10-F12)
+                Key.F10 => "F10",
+                Key.F11 => "F11",
+                Key.F12 => "F12",
+                // Last resort, use Enum<Key>.ToString() if the range is defined
+                _ when IsDefinedKey(key) => key.ToString(),
+                // Everything else failed, we throw an exception
+                _ => throw GetConvertToException(value, destinationType)
+            };
         }
 
-        private object GetKey(string keyToken, CultureInfo culture)
+        /// <summary>
+        /// Helper function that performs the conversion of <paramref name="keyToken"/> to the <see cref="Key"/> enum.
+        /// </summary>
+        /// <param name="keyToken">The string to convert from.</param>
+        /// <returns>A <see cref="Key"/> value corresponding to the specified string, <see cref="Key.None"/> if <paramref name="keyToken"/> was empty.</returns>
+        private static Key GetKeyFromString(ReadOnlySpan<char> keyToken)
         {
-            if (keyToken.Length == 0)
-            {
+            // If the token is empty, we presume "None" as our value but it is a success
+            if (keyToken.IsEmpty)
                 return Key.None;
+
+            // In case we're dealing with a lowercase character, we uppercase it
+            char firstChar = keyToken[0];
+            if (firstChar >= 'a' && firstChar <= 'z')
+                firstChar ^= (char)0x20;
+
+            // If this is a single-character we're dealing with, match digits/letters
+            if (keyToken.Length == 1 && char.IsLetterOrDigit(firstChar))
+            {
+                // Match a digit (0-9) or an ASCII letter (lower/uppercase)
+                if (char.IsDigit(firstChar) && (firstChar >= '0' && firstChar <= '9'))
```

Comment:
You can use:
```suggestion
                if (char.IsAsciiDigit(firstChar))
```

Reply:
This makes more sense and will be faster, thanks, will fix.